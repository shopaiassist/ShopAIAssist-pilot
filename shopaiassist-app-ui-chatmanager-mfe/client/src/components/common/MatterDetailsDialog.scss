.matter-details-dialog {
   .matter-details-dialog-body {
      min-width: 35rem;
      display: flex;
      flex-direction: column;
      gap: 1rem;
      padding-bottom: 0.25rem;
   }

   .matter-details-dialog-footer {
      display: flex;
      justify-content: flex-end;
      gap: 1rem;
   }
}
