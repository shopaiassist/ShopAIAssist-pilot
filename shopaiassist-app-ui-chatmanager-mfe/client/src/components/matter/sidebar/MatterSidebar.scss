@import "../../../utils/styles/imports";

$headerHeight: 3.75rem;

.matter-sidebar-container {
   display: flex;
   width: 21rem;
   z-index: map-get($z-index-map, "default");

   &.compact-mode {
      position:absolute;
      width: 100%;
   }

   &.full-mode {
      position:relative;
      width: 21rem;
      max-width: 100%;
   }

   .matter-sidebar-nav-header {
      display: flex;
      padding: 0.5rem;
      justify-content: space-between;
      align-items: center;
      align-self: stretch;
      min-height: 3.688rem;
      box-sizing: border-box;
      width: 100%;
      border: rem-calc(1);
      border-bottom: $saf-line-width-thin solid $saf-color-line-subtle;
      border-left: $saf-line-width-thin solid $saf-color-line-subtle;

      .matter-sidebar-nav-back {
         display: flex;
         gap: 0.5rem;
      }
   }

   .matter-sidebar {
      display: flex;
      flex-direction: column;
      align-items: center;
      box-shadow: $saf-drop-shadow-level-3;
      height: calc(100vh - $headerHeight);
      width: 21rem;
      background-color: $saf-color-background-subtle;

      &.compact-mode {
         width: 100%;
      }

      &.full-mode {
         width: 21rem;
         max-width: 100%;
      }

      .matter-sidebar-folder-name {
         padding: 0.5rem;
         margin: 0.75rem;
         display: flex;
         align-self: flex-start;
      }

      .matter-sidebar-header {
         display: flex;
         flex-direction: row;
         justify-content: space-between;
         width: 95%;
         padding: 0.5rem;
      }

      .matter-sidebar-footer {
         display: flex;
         width: 100%;
      }
   }
}