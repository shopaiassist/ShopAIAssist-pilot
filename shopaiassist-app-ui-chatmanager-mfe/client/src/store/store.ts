import { Action, AsyncThunkPayloadCreator, configureStore, ThunkAction } from '@reduxjs/toolkit';

import userReducer from './userSlice';

export const reducer = {
  user: userReducer
};

export const store = configureStore({
  reducer
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;

export interface AppThunkApiConfig {
  dispatch: AppDispatch;
  rejectValue: string;
  state: RootState;
}
export type AppAsyncThunk<P, R> = AsyncThunkPayloadCreator<R, P, AppThunkApiConfig>;
export type AppThunk<ReturnType = void> = ThunkAction<ReturnType, RootState, unknown, Action<string>>;
