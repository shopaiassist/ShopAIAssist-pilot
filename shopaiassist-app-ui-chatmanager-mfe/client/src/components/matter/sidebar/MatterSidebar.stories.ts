// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
import MatterSidebar from './MatterSidebar';
import { Meta, StoryObj } from '@storybook/react';

const meta = {
  title: 'Matter/Sidebar',
  component: MatterSidebar,
  parameters: {
    layout: 'padded'
  },
  tags: ['autodocs']
} satisfies Meta<typeof MatterSidebar>;

export default meta;

export const Default: StoryObj = { args: {} };
