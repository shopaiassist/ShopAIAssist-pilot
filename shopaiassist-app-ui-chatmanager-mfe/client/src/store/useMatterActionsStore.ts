import { create } from 'zustand';
import { devtools } from 'zustand/middleware';
import axios from 'axios';
import { debounce } from 'lodash';
import { FolderItem } from '../@types/sidebar';
import { useChatSidebarStore } from './useChatSidebarStore';
import { Endpoints } from '../utils/api';
import { useMfeContextUtilsStore } from './useMfeContextUtilsStore';
import i18nInstance from '../utils/i18n';
import { useBreadCrumbStore } from './useBreadcrumbStore';

export enum MatterDialogState {
  NEW_MATTER = 'new',
  DELETE_MATTER = 'delete',
  ARCHIVE_MATTER = 'archive',
  UNARCHIVE_MATTER = 'unarchive',
  UPDATE_MATTER = 'update'
}

type DialogState =
  | MatterDialogState.NEW_MATTER
  | MatterDialogState.DELETE_MATTER
  | MatterDialogState.ARCHIVE_MATTER
  | MatterDialogState.UNARCHIVE_MATTER
  | MatterDialogState.UPDATE_MATTER;

/**
 * Defines the structure and types of the state managed in the MatterActionsStore.
 * @interface MatterActionsState
 * @property {FolderItem|undefined} actionItem - The item currently being acted upon in the dialog.
 * @property {boolean} hideMatterActionsDialog - Indicates whether the matter actions dialog is hidden.
 * @property {DialogState} dialogState - The current state of the dialog (new, delete, archive, or update).
 * @property {Function} toggleMatterDialog - Function to toggle the visibility and state of the matter actions dialog.
 * @property {Function} createNewMatter - Function to create a new matter folder.
 * @property {Function} deleteFolder - Function to delete a folder by its identifier.
 * @property {Function} archiveFolder - Function to archive a folder by its identifier.
 * @property {Function} updateFolder - Function to update a folder by its identifier.
 */
export interface MatterActionsState {
  actionItem?: FolderItem;
  hideMatterActionsDialog: boolean;
  dialogState: DialogState;
  toggleMatterDialog: (dialogState?: DialogState, actionItem?: FolderItem) => void;
  createNewMatter: (folderInfo: Partial<FolderItem>) => void;
  deleteFolder: (folderId: string) => void;
  archiveFolder: (folderId: string) => void;
  unarchiveFolder: (folderId?: string) => void;
  updateFolder: (folderId: string, updates: Partial<FolderItem>) => void;
}

/**
 * The base URL constructor for API endpoints, utilizes the environment variable for the app's domain.
 * @param {string} endpoint - The API endpoint to be appended to the base domain.
 * @returns {string} The full URL for the API call.
 */
const BASE_URL = (endpoint: string) => `${process.env.APP_DOMAIN}/api/${endpoint}`;

/**
 * Zustand store to manage the state and actions related to matter creation, deletion, archiving, and dialog visibility.
 *
 * @returns {MatterActionsState} The state and actions for managing matter-related operations.
 */
export const useMatterActionsStore = create<MatterActionsState>()(
  devtools(
    (set, get) => ({
      actionItem: undefined,
      dialogState: MatterDialogState.NEW_MATTER,
      hideMatterActionsDialog: true,
      /**
       * Creates a new matter folder and refreshes the chat list.
       * Uses a debounced function to limit the rate of API calls.
       *
       * @param {Partial<FolderItem>} folderInfo - Partial information about the folder to be created.
       */
      createNewMatter: debounce(async (folderInfo: Partial<FolderItem>) => {
        await axios.post(BASE_URL(Endpoints.FOLDERS), { ...folderInfo }).then((response) => {
          useChatSidebarStore.getState().fetchChatList();
          useChatSidebarStore.getState().setActiveFolder(response.data.newFolder as FolderItem);
          set({
            hideMatterActionsDialog: !get().hideMatterActionsDialog,
            dialogState: undefined,
            actionItem: undefined
          });
        });
      }),
      /**
       * Deletes a folder by its identifier and refreshes the chat list.
       * Uses a debounced function to limit the rate of API calls.
       *
       * @param {string} folderId - The identifier of the folder to be deleted.
       */
      deleteFolder: debounce(async (folderId: string) => {
        const deleteFolderResponse = await axios.delete(BASE_URL(Endpoints.FOLDERS), { data: { id: folderId } });
        if (deleteFolderResponse.status === 200) {
          useMfeContextUtilsStore.getState().triggerAlert(i18nInstance.t('MATTER_DELETE_SUCCESS'), 'success');
        } else {
          useMfeContextUtilsStore.getState().triggerAlert(i18nInstance.t('ERROR_MESSAGE'), 'error');
        }
        useChatSidebarStore.getState().fetchChatList();
        set({
          hideMatterActionsDialog: !get().hideMatterActionsDialog,
          dialogState: undefined,
          actionItem: undefined
        });
      }),
      /**
       * Archives a folder by its identifier and refreshes the chat list.
       * Uses a debounced function to limit the rate of API calls.
       *
       * @param {string} folderId - The identifier of the folder to be archived.
       */
      archiveFolder: debounce(async (folderId: string) => {
        const archiveFolderResponse = await axios.patch(BASE_URL(Endpoints.FOLDERS), {
          id: folderId,
          updates: { isArchived: true }
        });
        if (archiveFolderResponse.status === 200) {
          useMfeContextUtilsStore.getState().triggerAlert(i18nInstance.t('MATTER_ARCHIVE_SUCCESS'), 'success');
        } else {
          useMfeContextUtilsStore.getState().triggerAlert(i18nInstance.t('ERROR_MESSAGE'), 'error');
        }
        useChatSidebarStore.getState().fetchChatList();
        set({
          hideMatterActionsDialog: !get().hideMatterActionsDialog,
          dialogState: undefined,
          actionItem: undefined
        });
      }),
      /**
       * Unarchives a folder by its identifier and refreshes the chat list.
       * Uses a debounced function to limit the rate of API calls.
       *
       * @param {string} folderId - The identifier of the folder to be archived.
       */
      unarchiveFolder: debounce(async (folderId?: string) => {
        if (!folderId) {
          useMfeContextUtilsStore.getState().triggerAlert(i18nInstance.t('ERROR_MESSAGE'), 'error');
          return;
        }

        useBreadCrumbStore.getState().removeBreadcrumb('archived_matters');
        useChatSidebarStore.getState().toggleArchivedMattersView();

        const unarchiveFolderResponse = await axios.patch(BASE_URL(Endpoints.FOLDERS), {
          id: folderId,
          updates: { isArchived: false }
        });
        if (unarchiveFolderResponse.status === 200) {
          useMfeContextUtilsStore.getState().triggerAlert(i18nInstance.t('MATTER_UNARCHIVE_SUCCESS'), 'success');
        } else {
          useMfeContextUtilsStore.getState().triggerAlert(i18nInstance.t('ERROR_MESSAGE'), 'error');
        }
        useChatSidebarStore.getState().fetchChatList();
        useChatSidebarStore.getState().refreshActiveFolder();
        set({
          dialogState: undefined,
          actionItem: undefined
        });
      }),
      /**
       * Updates a folder by its identifier and refreshes the chat list.
       * Uses a debounced function to limit the rate of API calls.
       *
       * @param {string} folderId - The identifier of the folder to be updated.
       * @param {Partial<FolderItem>} updates - The updates to apply to the folder.
       */
      updateFolder: debounce(async (folderId: string, updates: Partial<FolderItem>) => {
        await axios.patch(BASE_URL(Endpoints.FOLDERS), { id: folderId, updates }).then(() => {
          const isMatterSettingsActive = useChatSidebarStore.getState().isMatterSettingsActive;
          const hideMatterActionsDialog = get().hideMatterActionsDialog;
          if (updates.name) {
            useBreadCrumbStore.getState().updateBreadcrumb(folderId, updates.name);
          }

          useChatSidebarStore.getState().fetchChatList();

          if (useChatSidebarStore.getState().activeFolder) {
            useChatSidebarStore.getState().refreshActiveFolder();
          }

          set({
            hideMatterActionsDialog: isMatterSettingsActive ? hideMatterActionsDialog : !hideMatterActionsDialog,
            dialogState: undefined,
            actionItem: undefined
          });
        });
      }),
      /**
       * Toggles the visibility and state of the matter actions dialog.
       *
       * @param {DialogState} [dialogState] - The state of the dialog (new or delete).
       * @param {FolderItem} [actionItem] - The item currently being acted upon in the dialog.
       */
      toggleMatterDialog: (dialogState?: DialogState, newActionItem?: FolderItem) => {
        const currentDialogState = get().hideMatterActionsDialog;
        set({
          hideMatterActionsDialog: !currentDialogState,
          dialogState,
          actionItem: currentDialogState ? newActionItem : undefined
        });
      }
    }),
    { name: 'useMatterActionsStore' }
  )
);
