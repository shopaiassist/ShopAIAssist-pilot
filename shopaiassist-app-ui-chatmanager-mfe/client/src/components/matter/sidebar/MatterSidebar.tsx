import { Text } from '@on/core-components/react';
import MatterSidebarHeader from './header/MatterSidebarHeader';
import ChatList from './chat-list/ChatList';
import { FolderItem, TreeItem } from '../../../@types/sidebar';
import MatterSidebarNavigation from './header/MatterSidebarNavigation';
import useMeasure from 'react-use-measure';
import { animated, useSpring } from '@react-spring/web';
import { useChatSidebarStore } from '../../../store/useChatSidebarStore';
import MatterSidebarFooter from './footer/MatterSidebarFooter';

import './MatterSidebar.scss';
import classNames from 'classnames';
import { ModalType } from '../../../App';
import { useMatters } from '../../../views/Context';

interface MatterSidebarProps {
  isOpen: boolean;
  tree?: (TreeItem | FolderItem)[];
  onNewChat: () => void;
  closeSidebar: () => void;
}

/**
 * A responsive sidebar component for displaying a tree of items such as chats and folders.
 * This sidebar can be toggled to show or hide using an animated transition.
 * The component adapts to changes in the sidebar's width and employs animations for smooth transitions.
 *
 * @param {MatterSidebarProps} props - The properties for the MatterSidebar component.
 * @param {(TreeItem | FolderItem)[]} [props.tree] - Data structure representing the tree items to be displayed in the sidebar.
 * @param {() => void} props.onNewChat - Callback function that triggers when a new chat is initiated.
 * @returns {JSX.Element} The animated sidebar containing a header and a list of chat or folder items.
 *
 * @example
 * <MatterSidebar tree={treeData} onNewChat={handleNewChat} />
 */
const MatterSidebar = ({ isOpen, tree, onNewChat, closeSidebar }: MatterSidebarProps) => {
  const [measureRef, { width }] = useMeasure();
  const springProps = useSpring({
    transform: !isOpen ? `translateX(-${width}px)` : 'translateX(0px)',
    marginRight: !isOpen ? `-${width}px` : '0px'
  });
  const [
    activeFolder,
    setIsMattersettingsActive,
    setIsMatterFilesActive,
    setActiveChat,
    archivedMattersView,
    toggleArchivedMattersView
  ] = useChatSidebarStore((state) => [
    state.activeFolder,
    state.setIsMattersettingsActive,
    state.setIsMatterFilesActive,
    state.setActiveChat,
    state.archivedMattersView,
    state.toggleArchivedMattersView
  ]);
  const { modalType } = useMatters();

  return (
    <animated.div
      style={springProps}
      className={classNames('matter-sidebar-container', {
        'compact-mode': modalType === ModalType.Compact,
        'full-mode': modalType !== ModalType.Compact
      })}
    >
      <div
        ref={measureRef}
        className={classNames('matter-sidebar', {
          'compact-mode': modalType === ModalType.Compact,
          'full-mode': modalType !== ModalType.Compact
        })}
        data-testid="matter-sidebar"
        is-open={isOpen.toString()}
      >
        <MatterSidebarNavigation
          className="matter-sidebar-nav-header"
          isSubsectionView={!!activeFolder || !!archivedMattersView}
          closeSidebar={closeSidebar}
        />
        {activeFolder && (
          <div className="matter-sidebar-folder-name">
            <Text appearance="heading-lg">{activeFolder?.name}</Text>
          </div>
        )}
        {archivedMattersView && (
          <div className="matter-sidebar-folder-name">
            <Text appearance="heading-lg">Archived Matters</Text>
          </div>
        )}
        <MatterSidebarHeader
          className="matter-sidebar-header"
          onNewChat={() => {
            onNewChat();
            if (modalType === ModalType.Compact) {
              closeSidebar();
            }
          }}
          showAddChatButton={!archivedMattersView}
          isFolderView={!!activeFolder}
        />
        <ChatList chatsAndFolders={tree} closeSidebar={closeSidebar} />

        {/* <MatterSidebarFooter
          activeFolder={activeFolder}
          archivedMattersView={archivedMattersView}
          onOpenSettings={() => {
            setIsMattersettingsActive(true), setIsMatterFilesActive(false);
          }}
          onOpenMatterFiles={() => {
            setActiveChat();
            setIsMatterFilesActive(true);
          }}
          toggleArchivedMattersView={toggleArchivedMattersView}
        /> */}
      </div>
    </animated.div>
  );
};

export default MatterSidebar;
