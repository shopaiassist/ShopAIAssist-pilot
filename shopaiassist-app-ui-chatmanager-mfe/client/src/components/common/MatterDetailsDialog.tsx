import { Dialog, Button, Textfield, Textarea, Text, Alert } from '@on/core-components/react';
import Translate from '../i18n/Translate';
import { useTranslation } from 'react-i18next';
import { UpdateType, useMatterDetailsDialog } from '../../hooks/useMatterDetailsDialog';
import { MatterDialogState, useMatterActionsStore } from '../../store/useMatterActionsStore';

import './MatterDetailsDialog.scss';

/**
 * MatterDetailsDialog component provides a dialog interface for creating or updating matter details.
 *
 * The dialog includes input fields for matter name, description, and ID, with validation and state management
 * provided through custom hooks. It utilizes Dialog, Button, Textfield, and Textarea components
 * from the '@on/core-components/react' library for its UI elements.
 *
 * @component
 * @example
 * return (
 *   <MatterDetailsDialog />
 * )
 */
const MatterDetailsDialog = () => {
  const { t } = useTranslation();
  const [dialogState, hideMatterActionsDialog, toggleMatterDialog] = useMatterActionsStore((state) => [
    state.dialogState,
    state.hideMatterActionsDialog,
    state.toggleMatterDialog
  ]);
  const [matterName, matterId, matterDescription, dialogCopy, updateMatterDetails, dialogAction] =
    useMatterDetailsDialog();

  return (
    <Dialog
      hidden={hideMatterActionsDialog}
      dialogTitle={t(dialogCopy.title)}
      isFooter="true"
      onClose={() => toggleMatterDialog(MatterDialogState.NEW_MATTER, undefined)}
      className="matter-details-dialog"
    >
      <div className="matter-details-dialog-body">
        {dialogState === MatterDialogState.DELETE_MATTER && (
          <Alert hideCloseButton={true} appearance="error">
            {t(dialogCopy.subTitle)}
          </Alert>
        )}
        <Text appearance="heading-md">{`* ${t('INDICATES_REQUIRED_FIELD')}`}</Text>
        <Textfield
          data-testid="matter-dialog-name-field"
          label={`${t('NAME')}*`}
          maxlength={128}
          value={matterName}
          disabled={dialogState === MatterDialogState.DELETE_MATTER || dialogState === MatterDialogState.ARCHIVE_MATTER}
          onInput={(e) => {
            updateMatterDetails(UpdateType.NAME, e.target.value);
          }}
        />
        <Textarea
          data-testid="matter-dialog-description-field"
          label={t('DESCRIPTION')}
          maxlength={2000}
          resize="vertical"
          value={matterDescription}
          disabled={dialogState === MatterDialogState.DELETE_MATTER || dialogState === MatterDialogState.ARCHIVE_MATTER}
          onInput={(e) => {
            updateMatterDetails(UpdateType.DESCRIPTION, e.target.value);
          }}
        />
        <Textfield
          data-testid="matter-dialog-id-field"
          label={t('MATTER_ID')}
          maxlength={128}
          value={matterId}
          disabled={dialogState === MatterDialogState.DELETE_MATTER || dialogState === MatterDialogState.ARCHIVE_MATTER}
          onChange={(e) => {
            updateMatterDetails(UpdateType.ID, e.target.value);
          }}
        />
      </div>
      <div slot="footer" className="matter-details-dialog-footer">
        <Button
          appearance="secondary"
          onClick={() => toggleMatterDialog(MatterDialogState.NEW_MATTER, undefined)}
          data-testid="matter-dialog-close"
        >
          <Translate tKey={'CANCEL'} />
        </Button>
        <Button
          disabled={!matterName.length}
          appearance="primary"
          onClick={dialogAction}
          data-testid="matter-dialog-confirm"
        >
          <Translate tKey={dialogCopy.actionButton} />
        </Button>
      </div>
    </Dialog>
  );
};

export default MatterDetailsDialog;
