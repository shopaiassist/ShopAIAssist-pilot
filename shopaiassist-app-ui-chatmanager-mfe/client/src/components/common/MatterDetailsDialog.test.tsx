import axios from 'axios';
import { act, fireEvent, renderWithI18n, snapshot, waitFor } from '../../../jest-utils';
import { MatterDialogState, useMatterActionsStore } from '../../store/useMatterActionsStore';
import MatterDetailsDialog from './MatterDetailsDialog';
import { mockTree } from '../../utils/test-utils/mockData';
import { FolderItem } from '../../@types/sidebar';

const getMockedComponent = () => {
  return <MatterDetailsDialog />;
};

jest.mock('axios');
const mockedAxios = axios as jest.Mocked<typeof axios>;
const mockedMatter: FolderItem = {
  treeItemId: 'fake-item-id',
  name: 'fakeMatter',
  createdAt: new Date(),
  updatedAt: new Date(),
  type: 'folder',
  description: 'This is a fake matter',
  matterId: 'fake-custom-matter-id',
  fileCollectionId: 'whatever'
};

describe('MatterDetailsDialog Tests', () => {
  const store = useMatterActionsStore.getState();

  beforeEach(() => {
    useMatterActionsStore.setState(store, true);
    jest.clearAllMocks();
  });

  describe('render', () => {
    it('should render new matter', async () => {
      store.toggleMatterDialog(MatterDialogState.NEW_MATTER);
      snapshot(getMockedComponent());
    });
    it('should render delete matter', async () => {
      store.toggleMatterDialog(MatterDialogState.DELETE_MATTER, mockedMatter);
      snapshot(getMockedComponent());
    });
    it('should render archive matter', async () => {
      store.toggleMatterDialog(MatterDialogState.ARCHIVE_MATTER, mockedMatter);
      snapshot(getMockedComponent());
    });
  });

  describe('user action', () => {
    it('should call to create a new matter', async () => {
      store.toggleMatterDialog(MatterDialogState.NEW_MATTER);
      mockedAxios.get.mockResolvedValue(mockTree);
      mockedAxios.post.mockResolvedValue({ status: 201, data: { newFolder: mockedMatter } });
      const component = getMockedComponent();
      const wrapper = renderWithI18n(component);
      const confirm = wrapper.getByTestId('matter-dialog-confirm');
      expect(confirm).toBeDisabled();

      const nameField = wrapper.getByTestId('matter-dialog-name-field');
      expect(nameField).not.toBeDisabled();
      fireEvent.input(nameField, { target: { value: 'new name folder 1' } });
      expect(confirm).not.toBeDisabled();

      const descField = wrapper.getByTestId('matter-dialog-description-field');
      expect(descField).not.toBeDisabled();
      fireEvent.input(descField, { target: { value: 'new description folder 1' } });

      const matterIdField = wrapper.getByTestId('matter-dialog-id-field');
      expect(matterIdField).not.toBeDisabled();
      fireEvent.input(matterIdField, { target: { value: 'new matter id folder 1' } });

      confirm.click();
      await act(() => waitFor(() => wrapper.rerender(component)));
      expect(mockedAxios.post).toHaveBeenCalledWith('jest-test-domain/api/folders', {
        name: 'new name folder 1',
        description: 'new description folder 1',
        matterId: 'new matter id folder 1'
      });
    });

    it('should call to delete a matter', async () => {
      store.toggleMatterDialog(MatterDialogState.DELETE_MATTER, mockedMatter);
      mockedAxios.get.mockResolvedValue(mockTree);
      mockedAxios.delete.mockResolvedValue({ status: 200 });
      const component = getMockedComponent();
      const wrapper = renderWithI18n(component);

      const nameField = wrapper.getByTestId('matter-dialog-name-field');
      expect(nameField).toBeDisabled();

      const descField = wrapper.getByTestId('matter-dialog-description-field');
      expect(descField).toBeDisabled();

      const matterIdField = wrapper.getByTestId('matter-dialog-id-field');
      expect(matterIdField).toBeDisabled();

      const confirm = wrapper.getByTestId('matter-dialog-confirm');
      confirm.click();

      await act(() => waitFor(() => wrapper.rerender(component)));
      expect(mockedAxios.delete).toHaveBeenCalledWith('jest-test-domain/api/folders', {
        data: { id: mockedMatter.treeItemId }
      });
    });

    it('should call to archive a matter', async () => {
      store.toggleMatterDialog(MatterDialogState.ARCHIVE_MATTER, mockedMatter);
      mockedAxios.get.mockResolvedValue(mockTree);
      mockedAxios.patch.mockResolvedValue({ status: 200 });
      const component = getMockedComponent();
      const wrapper = renderWithI18n(component);

      const nameField = wrapper.getByTestId('matter-dialog-name-field');
      expect(nameField).toBeDisabled();

      const descField = wrapper.getByTestId('matter-dialog-description-field');
      expect(descField).toBeDisabled();

      const matterIdField = wrapper.getByTestId('matter-dialog-id-field');
      expect(matterIdField).toBeDisabled();

      const confirm = wrapper.getByTestId('matter-dialog-confirm');
      confirm.click();

      await act(() => waitFor(() => wrapper.rerender(component)));
      expect(mockedAxios.patch).toHaveBeenCalledWith('jest-test-domain/api/folders', {
        id: mockedMatter.treeItemId,
        updates: { isArchived: true }
      });
    });

    it('should call to update a matter', async () => {
      store.toggleMatterDialog(MatterDialogState.UPDATE_MATTER, mockedMatter);
      mockedAxios.get.mockResolvedValue(mockTree);
      mockedAxios.patch.mockResolvedValue({ status: 200 });
      const component = getMockedComponent();
      const wrapper = renderWithI18n(component);
      const confirm = wrapper.getByTestId('matter-dialog-confirm');
      expect(confirm).not.toBeDisabled();

      const nameField = wrapper.getByTestId('matter-dialog-name-field');
      expect(nameField).not.toBeDisabled();
      fireEvent.input(nameField, { target: { value: 'Changed Name' } });

      const descField = wrapper.getByTestId('matter-dialog-description-field');
      expect(descField).not.toBeDisabled();
      fireEvent.input(descField, { target: { value: 'Changed Description' } });

      const matterIdField = wrapper.getByTestId('matter-dialog-id-field');
      expect(matterIdField).not.toBeDisabled();
      fireEvent.input(matterIdField, { target: { value: 'Changed Matter ID' } });

      confirm.click();
      await act(() => waitFor(() => wrapper.rerender(component)));
      expect(mockedAxios.patch).toHaveBeenCalledWith('jest-test-domain/api/folders', {
        id: mockedMatter.treeItemId,
        updates: { description: 'Changed Description', matterId: 'Changed Matter ID', name: 'Changed Name' }
      });
    });

    it('should close when user clicks cancel', async () => {
      store.toggleMatterDialog(MatterDialogState.NEW_MATTER);
      const component = getMockedComponent();
      const wrapper = renderWithI18n(component);
      const confirm = wrapper.getByTestId('matter-dialog-close');
      confirm.click();
      await act(() => waitFor(() => wrapper.rerender(component)));
      expect(store.hideMatterActionsDialog).toBe(true);
    });
  });
});
