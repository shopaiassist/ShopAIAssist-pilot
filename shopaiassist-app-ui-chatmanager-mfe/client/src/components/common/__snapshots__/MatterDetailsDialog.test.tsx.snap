// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatterDetailsDialog Tests render should render archive matter 1`] = `
<div
  class="matter-details-dialog"
  dialogtitle="Archive matter?"
  isfooter="true"
>
  <div
    class="matter-details-dialog-body"
  >
    <span
      appearance="heading-md"
    >
      * indicates required field
    </span>
    <textarea
      data-testid="matter-dialog-name-field"
      disabled=""
      label="Name*"
      maxlength="128"
    >
      fakeMatter
    </textarea>
    <textarea
      data-testid="matter-dialog-description-field"
      disabled=""
      label="Description"
      maxlength="2000"
      resize="vertical"
    >
      This is a fake matter
    </textarea>
    <textarea
      data-testid="matter-dialog-id-field"
      disabled=""
      label="Matter ID"
      maxlength="128"
    >
      fake-custom-matter-id
    </textarea>
  </div>
  <div
    class="matter-details-dialog-footer"
    slot="footer"
  >
    <button
      appearance="secondary"
      data-testid="matter-dialog-close"
    >
      Cancel
    </button>
    <button
      appearance="primary"
      data-testid="matter-dialog-confirm"
    >
      Archive matter
    </button>
  </div>
</div>
`;

exports[`MatterDetailsDialog Tests render should render delete matter 1`] = `
<div
  class="matter-details-dialog"
  dialogtitle="Delete matter?"
  isfooter="true"
>
  <div
    class="matter-details-dialog-body"
  >
    <span
      appearance="error"
    >
      All chats and files in the matter will be lost. This can't be undone.
    </span>
    <span
      appearance="heading-md"
    >
      * indicates required field
    </span>
    <textarea
      data-testid="matter-dialog-name-field"
      disabled=""
      label="Name*"
      maxlength="128"
    >
      fakeMatter
    </textarea>
    <textarea
      data-testid="matter-dialog-description-field"
      disabled=""
      label="Description"
      maxlength="2000"
      resize="vertical"
    >
      This is a fake matter
    </textarea>
    <textarea
      data-testid="matter-dialog-id-field"
      disabled=""
      label="Matter ID"
      maxlength="128"
    >
      fake-custom-matter-id
    </textarea>
  </div>
  <div
    class="matter-details-dialog-footer"
    slot="footer"
  >
    <button
      appearance="secondary"
      data-testid="matter-dialog-close"
    >
      Cancel
    </button>
    <button
      appearance="primary"
      data-testid="matter-dialog-confirm"
    >
      Delete matter
    </button>
  </div>
</div>
`;

exports[`MatterDetailsDialog Tests render should render new matter 1`] = `
<div
  class="matter-details-dialog"
  dialogtitle="Create matter folder"
  isfooter="true"
>
  <div
    class="matter-details-dialog-body"
  >
    <span
      appearance="heading-md"
    >
      * indicates required field
    </span>
    <textarea
      data-testid="matter-dialog-name-field"
      label="Name*"
      maxlength="128"
    />
    <textarea
      data-testid="matter-dialog-description-field"
      label="Description"
      maxlength="2000"
      resize="vertical"
    />
    <textarea
      data-testid="matter-dialog-id-field"
      label="Matter ID"
      maxlength="128"
    />
  </div>
  <div
    class="matter-details-dialog-footer"
    slot="footer"
  >
    <button
      appearance="secondary"
      data-testid="matter-dialog-close"
    >
      Cancel
    </button>
    <button
      appearance="primary"
      data-testid="matter-dialog-confirm"
      disabled=""
    >
      Create matter folder
    </button>
  </div>
</div>
`;
